### PR TITLE
Feature/#64 update last seen in db

### LINKED ISSUE #64: Update last_seen in db whenever user interacts with system
In the database table session the variable last_seen should be updated whenever the user interacts with the system.

### PR DESCRIPTION
**Pull request by McPoppe**
Fixes #64

### CODE REVIEW

File: src/main/java/se/lth/base/server/rest/UserResource.java [hunk 1]
```diff
@@ -137,4 +137,19 @@ public void deleteUser(@PathParam("id") int userId) {
             throw new WebApplicationException("You are not permitted to delete this user", Response.Status.FORBIDDEN);
         }
     }
+    
+    @Path("updateSessionLastSeen")
```
Comment (jost95) [inline src/main/java/se/lth/base/server/rest/UserResource.java]: This should not be its own rest call, we should add it in the authentication provider instead...

File: src/main/java/se/lth/base/server/rest/UserResource.java [hunk 2]
```diff
@@ -137,4 +137,19 @@ public void deleteUser(@PathParam("id") int userId) {
...
+    @PUT
+    @PermitAll
+    public void updateSessionLastSeen() {
+    	userDao.updateSessionLastSeen(session.getSessionId());
+    }
+    
+    @Path("session")
```
Comment (jost95) [inline src/main/java/se/lth/base/server/rest/UserResource.java]: This is not needed?

File: src/test/java/se/lth/base/server/rest/UserResourceTest.java [hunk 3]
```diff
@@ -128,45 +132,15 @@ public void createUserAsUser() {
                 .post(Entity.json(""), Void.class); // Include response type to trigger exception
     }
 
-    @Test(expected = ForbiddenException.class)
```
Comment (jost95) [inline src/test/java/se/lth/base/server/rest/UserResourceTest.java]: Why is this deleted?

File: src/test/java/se/lth/base/server/rest/UserResourceTest.java [hunk 4]
```diff
@@ -128,45 +132,15 @@ public void createUserAsUser() {
...
-        logout();
         login(TEST_CREDENTIALS);
         target("user")
-                .path(Integer.toString(newUser.getId()))
+                .path(Integer.toString(ADMIN.getId()))
                 .request()
                 .delete(Void.class); // Include response type to trigger exception
     }
 
-    @Test(expected = NotFoundException.class)
```
Comment (jost95) [inline src/test/java/se/lth/base/server/rest/UserResourceTest.java]: Why is this deleted?

File: src/main/java/se/lth/base/server/rest/UserResource.java (hunk 1 above)
Comment (McPoppe) [reply src/main/java/se/lth/base/server/rest/UserResource.java]: Not sure what you refer to when you say the authentication provider.

File: src/main/java/se/lth/base/server/rest/UserResource.java (hunk 2 above)
Comment (McPoppe) [reply src/main/java/se/lth/base/server/rest/UserResource.java]: needed the getSession for my test

File: src/test/java/se/lth/base/server/rest/UserResourceTest.java (hunk 3 above)
Comment (McPoppe) [reply src/test/java/se/lth/base/server/rest/UserResourceTest.java]: The only thing i remember changing in the test files was that i added my test and removed the test that checked so that a user couldnt remove themselves. Because i know this test was removed in another issue but for some reason it was back and failing since it was now possible to delete yourself. I didn't delete this test or change any others, at least not intentionally.

**Comment by McPoppe** [comment]
After looking through the code again this functionallity may already be implemented. The method getSession in UserDataAccess already updates last_seen. This pull request and the issue should be removed. My bad will make sure to properly check existing code before suggesting a new issue in the future.
